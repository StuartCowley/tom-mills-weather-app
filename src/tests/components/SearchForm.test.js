import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const testFunction = () => {};
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        onSubmit={testFunction}
        searchText=""
        setSearchText={testFunction}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders button text correctly", () => {
    const { getByText } = render(
      <SearchForm
        onSubmit={testFunction}
        searchText=""
        setSearchText={testFunction}
      />
    );
    expect(getByText("Search")).toBeInstanceOf(HTMLButtonElement);
  });
});
