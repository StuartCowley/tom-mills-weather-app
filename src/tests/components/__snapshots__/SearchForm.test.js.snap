// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div>
    <input
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
